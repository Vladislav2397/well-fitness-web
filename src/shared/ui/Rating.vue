<template lang="pug">

.rating
    span._label(
        v-if="hasLabel"
        :class="{ 'rating__label--bold': isBold }"
    )
        template(
            v-if="$slots.default"
        ): slot
        template(
            v-else
        ) Рейтинг
    ._list
        ._star(
            v-for="i in 5"
            :key="i"
            :class="{ 'rating__star--fill' : i <= count }"
        )

</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

@Component
export default class Rating extends Vue {
    @Prop({ default: 0 }) readonly count?: 0 | 1 | 2 | 3 | 4 | 5

    @Prop({ default: true }) readonly hasLabel!: boolean

    @Prop() readonly isBold!: boolean
}

</script>

<style lang="scss">
.rating {
    font-size: toRem(14);
    line-height: toRem(20);

    vertical-align: middle;
    color: $secondary;

    &__list {
        display: inline-block;
        vertical-align: middle;
        color: #f99808;
    }

    &__label {
        &--bold {
            font-weight: 600;
        }
    }

    &__label + &__list {
        margin-left: toRem(6);
    }

    &__star {
        width: toRem(10);
        height: toRem(10);
        display: inline-block;
        border: 1px solid;

        &--fill {
            background-color: #f99808; // FIXME: make a var
        }

        & + & {
            margin-left: toRem(2);
        }
    }
}
</style>
