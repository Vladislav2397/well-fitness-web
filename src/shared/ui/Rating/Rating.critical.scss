.rating {
    font-size: toRem(14);
    line-height: toRem(20);

    vertical-align: middle;
    color: $secondary;

    &__list {
        display: inline-block;
        vertical-align: middle;
        color: #f99808;
    }

    &__label {
        &--bold {
            font-weight: 600;
        }
    }

    &__label + &__list {
        margin-left: toRem(6);
    }

    &__star {
        width: toRem(10);
        height: toRem(10);
        display: inline-block;
        border: 1px solid;

        &--fill {
            background-color: #f99808; // FIXME: make a var
        }

        & + & {
            margin-left: toRem(2);
        }
    }
}
