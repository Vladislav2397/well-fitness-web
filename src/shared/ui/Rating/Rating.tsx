import { RenderContext } from 'vue'
import * as tsx from 'vue-tsx-support'

export type RatingProps = {
    isBold?: boolean
    hasLabel?: boolean
    count?: 0 | 1 | 2 | 3 | 4 | 5
}

type RatingStarProps = {
    isFill?: boolean
}

const RatingStarComponent = ({
    props: { isFill = false },
}: RenderContext<RatingStarProps>) => {
    return (
        <div
            class={[
                'rating__star',
                {
                    'rating__star--fill': isFill,
                },
            ]}
        />
    )
}
const RatingStar = tsx
    .ofType<RatingStarProps>()
    .convert(RatingStarComponent as any)

export const RatingComponent = ({
    slots,
    props: { isBold = false, hasLabel = false, count = 0 },
}: RenderContext<RatingProps>) => {
    return (
        <div class="rating">
            {hasLabel && (
                <span
                    class={[
                        'rating__label',
                        {
                            'rating__label--bold': isBold,
                        },
                    ]}>
                    {slots().default ? <slot /> : 'Рейтинг'}
                </span>
            )}
            <div class="rating__list">
                {[1, 2, 3, 4, 5].map(i => (
                    <RatingStar key={i} isFill={i <= count} />
                ))}
            </div>
        </div>
    )
}

export const Rating = tsx.ofType<RatingProps>().convert(RatingComponent as any)
