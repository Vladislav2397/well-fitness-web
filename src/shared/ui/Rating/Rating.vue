<template lang="pug">

.b-rating
    span.__label(
        v-if="hasLabel"
        :class="{ 'rating__label--bold': isBold }"
    )
        template(
            v-if="$slots.default"
        ): slot
        template(
            v-else
        ) Рейтинг
    .__list
        .__star(
            v-for="i in 5"
            :key="i"
            :class="{ 'rating__star--fill' : i <= count }"
        )

</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

@Component
export default class Rating extends Vue {
    @Prop({ default: 0 }) readonly count?: 0 | 1 | 2 | 3 | 4 | 5

    @Prop({ default: true }) readonly hasLabel!: boolean

    @Prop() readonly isBold!: boolean
}

</script>

<!--<style lang="scss" src="./Rating.critical.scss"/>-->
<!--<style lang="scss" src="./Rating.main.scss"/>-->
