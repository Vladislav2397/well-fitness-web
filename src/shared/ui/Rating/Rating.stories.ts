import { Rating, type RatingProps } from './Rating'
import { Story } from '@storybook/vue'

export default {
    title: 'shared/Rating',
    component: Rating,
    argTypes: {},
}

const Template: Story<RatingProps> = (args, { argTypes }) => ({
    props: Object.keys(argTypes),
    components: { Rating },
    methods: {},
    template: '<Rating v-bind="$props"/>',
})

export const Default = Template.bind({})
Default.args = {
    count: 0,
    hasLabel: false,
    isBold: false,
}
