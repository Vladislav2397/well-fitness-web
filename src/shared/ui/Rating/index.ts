import { Rating, type RatingProps } from './Rating'

export { Rating, RatingProps }
