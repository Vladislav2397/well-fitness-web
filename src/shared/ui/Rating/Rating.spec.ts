import { shallowMount } from '@vue/test-utils'
import { Rating } from './Rating'

describe('Rating component.spec', () => {
    test.each([0, 1, 2, 3, 4, 5])('Snapshots for count=%s', count => {
        const wrapper = shallowMount(Rating, {
            propsData: {
                count,
            },
        })
        expect(wrapper).toMatchSnapshot()
    })
})
