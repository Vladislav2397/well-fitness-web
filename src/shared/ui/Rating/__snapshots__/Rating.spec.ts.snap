// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component.spec Snapshots for count=0 1`] = `
<div class="rating">
  <div class="rating__list">
    <div class="rating__star"></div>
    <div class="rating__star"></div>
    <div class="rating__star"></div>
    <div class="rating__star"></div>
    <div class="rating__star"></div>
  </div>
</div>
`;

exports[`Rating component.spec Snapshots for count=1 1`] = `
<div class="rating">
  <div class="rating__list">
    <div class="rating__star rating__star--fill"></div>
    <div class="rating__star"></div>
    <div class="rating__star"></div>
    <div class="rating__star"></div>
    <div class="rating__star"></div>
  </div>
</div>
`;

exports[`Rating component.spec Snapshots for count=2 1`] = `
<div class="rating">
  <div class="rating__list">
    <div class="rating__star rating__star--fill"></div>
    <div class="rating__star rating__star--fill"></div>
    <div class="rating__star"></div>
    <div class="rating__star"></div>
    <div class="rating__star"></div>
  </div>
</div>
`;

exports[`Rating component.spec Snapshots for count=3 1`] = `
<div class="rating">
  <div class="rating__list">
    <div class="rating__star rating__star--fill"></div>
    <div class="rating__star rating__star--fill"></div>
    <div class="rating__star rating__star--fill"></div>
    <div class="rating__star"></div>
    <div class="rating__star"></div>
  </div>
</div>
`;

exports[`Rating component.spec Snapshots for count=4 1`] = `
<div class="rating">
  <div class="rating__list">
    <div class="rating__star rating__star--fill"></div>
    <div class="rating__star rating__star--fill"></div>
    <div class="rating__star rating__star--fill"></div>
    <div class="rating__star rating__star--fill"></div>
    <div class="rating__star"></div>
  </div>
</div>
`;

exports[`Rating component.spec Snapshots for count=5 1`] = `
<div class="rating">
  <div class="rating__list">
    <div class="rating__star rating__star--fill"></div>
    <div class="rating__star rating__star--fill"></div>
    <div class="rating__star rating__star--fill"></div>
    <div class="rating__star rating__star--fill"></div>
    <div class="rating__star rating__star--fill"></div>
  </div>
</div>
`;
